pcdsdevices/epics/state.py

```diff
@@ -79,17 +81,38 @@ def __init__(self, prefix, *, read_attrs=None, name=None, **kwargs):
 
     @property
     def states(self):
+        """
+        A list of possible states
+        """
         return list(self._states.keys())
 
     @property
     def value(self):
+        """
+        Current state of the object
+        """
+        state_value = None
         for state, info in self._states.items():
+            #Gather signal
             state_signal = getattr(self, state)
+            #Get raw EPICS readback
             pv_value = state_signal.get(use_monitor=True)
-            state_value = info.get(pv_value, "unknown")
-            if state_value != "defer":
-                return state_value
-        return "unknown"
+            try:
+                #Associate readback with device state
+                signal_state = info[pv_value]
+                if signal_state != 'defer':
+                    if state_value:
+                        assert signal_state == state_value
+                    #Update state
+                    state_value = signal_state
+
+            #Handle unaccounted readbacks 
+            except (KeyError, AssertionError):
+                state_value = 'unknown'
+                break
+
+        #If all states deferred, report as unknown
+        return state_value or 'unknown'
```

Comment:
I don't think this change was strictly necessary, you could do everything with the old schema, but I think it's a good change because it makes setting up the states more intuitive and it makes it easier to catch inconsistent states.